### DISCUSSION TITLE
(&:)

### DISCUSSION THREAD

**Discussion by B3Y0N4** [19/classroom#2] [Ruby] [answered]
%w[foo bar].map(&:upcase) #=> ['FOO', 'BAR']
soru:  yukarıdaki kodda  bu (&:) tam olarak nasil çalişiyor ve nezaman kullanmamiz gerekiyor

**Comment by Ferbman** [comment, marked as answer]
` (&:upcase) `   `{|x| x.upcase }` in kısaltması diye biliyorum . `&` nın tam işlevini bilmiyorum ama &: değil ordaki : işareti `:upcaseye` bağlı

**Reply by B3Y0N4** [reply]
peki upcase metodu niye symbol türünden yazilmiş?

**Reply by roktas** [reply]
Bu bir kısayol.  Bunun anlatımı şu noktada zor, bu haliyle kabul edin şimdilik.  Ayrıntı isteyenler için örnek bir okuma: https://medium.com/@sologoubalex/parameter-with-ampersand-operator-in-ruby-6a6a7fd666d5

**Reply by B3Y0N4** [reply]
Tamamdir teşşekür edrim hocam.
